### ISSUE TITLE
CurrReport.PREVIEW is not work correctly

### ISSUE THREAD

**Issue by hgnamvn** [Microsoft/AL#1017]
Hi!

I created a test report to test function CurrReport.Preview.
But it isn't work correctly as the link: [CurrReport.Preview.](https://docs.microsoft.com/en-us/dynamics-nav/preview-function--report-)
The Print icon, Print Layout icon, Page Setup icon, and Save As icon are still displayed.

Report:   
     ![image](https://user-images.githubusercontent.com/24407031/33410185-9a958926-d5b1-11e7-8b86-59210da137ff.png)

Result:
     ![image](https://user-images.githubusercontent.com/24407031/33410357-6cb9a7f2-d5b2-11e7-8078-97c90ef37a5a.png)

It seems that the return value of CurrReport.Preview function is not correct.

**_Note:_**
- Version: November Update
- I reference from report 206_Sales - Invoice. It work correctly.

**Comment by DominikDitoIvosevic** [comment]
I'm not sure why the icons are displayed but in your code I think you have a bug with the semicolon after THEN.

Semicolon by itself means an empty statement so that's the one in IF and the message then gets displayed always as a statement after the IF.
I tried the same code on a report created in CSIDE and I get the same error.
I also see the same icons when previewing a report created from CSIDE.

When I invoke Preview in OnPreReport trigger instead of OnPostReport (like you did) then I also do not see the icons.

tldr; You should remove the semicolon and use a different trigger to get the desired behavior

**Comment by hgnamvn** [comment]
Thank DominikDitoIvosevic,

I have a bug about the semicolon after THEN.
I tested again, the return value of **CurrReport.Preview** function is **correct**.
But these icon are **still displayed**. **They should are hidden**.

1. I modify report again.
     ![image](https://user-images.githubusercontent.com/24407031/33464516-40f6c258-d675-11e7-8227-cfab0f29e61d.png)

2. Result:
     ![image](https://user-images.githubusercontent.com/24407031/33464694-374de6e0-d676-11e7-8742-05271f68af4b.png)

**Comment by DominikDitoIvosevic** [comment]
You should invoke CurrReport.PREVIEW in trigger OnPreReport and not on OnPostReport

**Comment by hgnamvn** [comment]
Actually, I tested with OnPreReport trigger, but the result is the same.

I checked standard report 206 - "Sales Invoice", this report use CurrReport.PREVIEW in trigger OnAfterGetRecord and OnPostDataItem. **It work well**.

So, I don't understand why my report work incorrectly

**Comment by DominikDitoIvosevic** [comment]
Thank you for your persistance. I have found the culprit and I'm working on fixing it

**Comment by hgnamvn** [comment]
Thank DominikDitoIvosevic,

And, please consider the related issue: [issues 837](https://github.com/Microsoft/AL/issues/837)
